redisinsight/api/src/modules/settings/repositories/local.agreements.repository.ts

```diff
@@ -14,15 +15,22 @@ export class LocalAgreementsRepository extends AgreementsRepository {
     super();
   }
 
-  async getOrCreate(): Promise<Agreements> {
+  async getOrCreate(
+    sessionMetadata: SessionMetadata,
+    defaultOptions: DefaultAgreementsOptions = {},
+  ): Promise<Agreements> {
     let entity = await this.repository.findOneBy({});
-
     if (!entity) {
       try {
-        entity = await this.repository.save(this.repository.create({ id: 1 }));
+        entity = await this.repository.save(
+          classToClass(AgreementsEntity, plainToInstance(Agreements, {
+            ...defaultOptions,
+            id: 1,
+          })),
+        );
       } catch (e) {
         if (e.code === 'SQLITE_CONSTRAINT') {
-          return this.getOrCreate();
+          return this.getOrCreate(sessionMetadata);
```

Comment:
If this is a retry, shouldn't it pass the options as well?

```suggestion
          return this.getOrCreate(sessionMetadata, defaultOptions);
```